lib/model/migrations/20230512-03-add-entity-source.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+/* Steps of this migration
+  1. add new table entity source
+  2. add source id col to entity def
+  3. populate rows of source table and add id in entity def
+  4. remove sub def id from entity def table
+*/
+
+const up = async (db) => {
+  await db.schema.createTable('entity_def_sources', (es) => {
```

Comment:
I should have noticed this earlier, I think there was a consensus that new migrations should be written without knex API

Reply:
Do we need any indices here? That can happen in a separate PR